apkid/rules/dex/packers.yara

```diff
@@ -386,3 +386,50 @@ rule crazy_dog_wrapper : packer
     is_dex
     and 2 of them
 }
+
+rule jsonpacker_dex : packer
+{
+   meta:
+     description = "JsonPacker"
+     sample       = "e23f0a124fdaba30c07a3c40011dd99240af081cec4cdfcb990c811126867e59"
+     author        = "Axelle Apvrille and Eduardo Novella"
+
+   strings:
+     /* typical XOR algo with junk operations */
+     $algo = {
```

Comment:
How do you know that these code chunks are not going to match with another packer/ apk?

Reply:
I've tested a collection of APKs I have to see if it matches well or not. 
Like any generic signature, there is a small possibility of a False Positive though.

Reply:
note I am not maching only the XOR but the junk operations the code adds around it. Another packer could do the same, true, but unlikely.

`bArr[i9]) + (((i12 / i12) / 1) ^ 1)) + (i12 % i12)) ^ bArr3[i9 % 1]);`